lib/storage/azure.js

```diff
@@ -170,34 +171,75 @@ module.exports = function() {
           if (err) {
             return cb(err);
           }
+
           blobSvcs.push({
             svc: svc,
             container: cluster.container || options.container
           });
+
           return cb();
         });
       }, callback);
     },
 
-    copyIn: function(localPath, path, options, callback) {
-      // TODO be more selective about when we do this
-      gzipme(localPath, false, 'best', function() {
-        path = path[0] === '/' ? path.slice(1) : path;
-        async.each(blobSvcs, function(blobSvc, cb) {
-          createContainerBlob(blobSvc, path, localPath, function(err) {
-            if (err) {
-              __log('! Cannot create blob in container : ' + err);
+    cleanupStreams: function (inputStream, outputStream, tempPath, err) {
+      async.parallel({
+        unlink: cb => {
+          removeLocalBlob(tempPath, cb);
```

Comment:
Will this throw an error if it doesn't exist yet? That needs to be caught without being treated as a new problem. Or you need to track whether it's supposed to exist yet and not try to remove it if it doesn't.